ledger/state.go

```diff
@@ -1074,6 +1262,52 @@ func (ls *LedgerState) GetCurrentPParams() lcommon.ProtocolParameters {
 	return ls.currentPParams
 }
 
+// Test helper methods - these should only be used in tests
+func (ls *LedgerState) SetConfig(config LedgerStateConfig) {
+	ls.config = config
+}
+
+func (ls *LedgerState) SetCurrentEra(era eras.EraDesc) {
+	ls.currentEra = era
+}
+
+func (ls *LedgerState) SetCurrentEpoch(epoch models.Epoch) {
+	ls.currentEpoch = epoch
+}
+
+func (ls *LedgerState) SetCurrentPParams(pparams lcommon.ProtocolParameters) {
+	ls.currentPParams = pparams
+}
+
+func (ls *LedgerState) GetCurrentEra() eras.EraDesc {
+	return ls.currentEra
+}
+
+func (ls *LedgerState) GetCurrentEpoch() models.Epoch {
+	return ls.currentEpoch
+}
+
+func (ls *LedgerState) LoadGenesisProtocolParametersPublic() (lcommon.ProtocolParameters, error) {
+	return ls.loadGenesisProtocolParameters()
+}
+
+func (ls *LedgerState) LoadEpochsPublic(txn *database.Txn) error {
+	return ls.loadEpochs(txn)
+}
+
+func (ls *LedgerState) LoadPParamsPublic() error {
+	return ls.loadPParams()
+}
+
+// ProcessTransactionCertificatesPublic is a test helper to expose processTransactionCertificates
+func (ls *LedgerState) ProcessTransactionCertificatesPublic(
+	txn *database.Txn,
+	blockPoint ocommon.Point,
+	certs []ledger.Certificate,
+) error {
+	return ls.processTransactionCertificates(txn, blockPoint, certs)
+}
```

Comment:
These functions can affect the ledger state outside of processing TXs from blocks, and we should probably not be exposing them outside the package

Reply:
For this, I am going to fix/handle it in the "unified certificate" PR when I inevitably need to rebase it on top of this code.